tutorials/Oracles/Oracles.ipynb

```diff
@@ -0,0 +1,1015 @@
+    "is Adj + Ctl {\n",
+    "    // ...\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### <span style=\"color:blue\">Demo 3</span>: Oracle Converter\n",
+    "* **Input:**\n",
+    "    * a marking oracle\n",
+    "* **Output:**\n",
+    "    * a phase oracle which takes a register as input and flips the phase of that register according to the function $f(x)$ that the marking oracle implements\n",
+    "\n",
+    "In this demo we will use the oracle converter to convert your previously implemented marking oracle, $U_{7,mark}$ into a phase oracle.\n",
+    "\n",
+    "> **You must have Exercises 2, 3 and 4 completed correctly for this demo to behave correctly!**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "open Microsoft.Quantum.Canon;\n",
+    "open Microsoft.Quantum.Intrinsic;\n",
+    "open Microsoft.Quantum.Diagnostics;\n",
+    "\n",
+    "function Oracle_Converter(markingOracle: ((Qubit[], Qubit) => Unit is Adj + Ctl)) : (Qubit[] => Unit is Adj + Ctl) {\n",
+    "    return Apply_Phase_Oracle(markingOracle, _);\n",
+    "}\n",
+    "\n",
+    "operation Oracle_Converter_Demo() : Unit {\n",
+    "    let divider = \"--------------------------------------------------------------------------------------------------\";\n",
+    "\n",
+    "    // allocate the state |000⟩\n",
+    "    using (register = Qubit[3]) {\n",
+    "        // convert to the equal superposition state\n",
+    "        ApplyToEachA(H, register);\n",
+    "        \n",
+    "        // dump the state\n",
+    "        Message(\"the equal superposition register before applying the phase oracle you implemented:\");\n",
+    "        DumpMachine();\n",
+    "        Message(divider);\n",
+    "        \n",
+    "        // apply the oracle you implemented:\n",
+    "        Phase_7_Oracle(register);\n",
+    "        \n",
+    "        // dump the state after application of the oracle\n",
+    "        Message(\"the equal superposition register after applying the phase oracle you implemented:\");\n",
+    "        DumpMachine();\n",
+    "        Message(divider);\n",
+    "        \n",
+    "        // reset the qubits for deallocation\n",
+    "        ResetAll(register);\n",
+    "    }\n",
+    "    \n",
+    "    // allocate the state |000⟩\n",
+    "    using (register = Qubit[3]) {\n",
+    "        // convert to the equal superposition state\n",
+    "        ApplyToEachA(H, register);\n",
+    "        \n",
+    "        // dump the state\n",
+    "        Message(\"the equal superposition register before applying the converted marking oracle:\");\n",
+    "        DumpMachine();\n",
+    "        Message(divider);\n",
+    "        \n",
+    "        // convert the marking oracle\n",
+    "        let Converted_7_Oracle = Oracle_Converter(Marking_7_Oracle);\n",
```

Comment:
I wouldn't use the extra `Oracle_Converter` call if we're just going to apply the new oracle right away; I'd just use `Apply_Phase_Oracle` (well, with its new name) on the register immediately.